client/src/components/SendMessage.js

```diff
@@ -36,36 +39,73 @@ class SendMessage extends React.Component {
     return;
   }
 
+  hideEmojiPicker() {
+      this.setState({
+          ...this.state,
+          emojiPickerActive : false
+      });
+  }
+  onKeyUp(event) {
+      if (this.state.emojiPickerActive){
+          if (event.which === 9 || event.which === 37 || event.which === 39) {
+              // Cycle through preview
+              this.refs.emojipicker.onKeyUp(event);
+          } else {
+              const lastWord = this.refs.message.value.split(" ").pop();
+              this.refs.emojipicker.filterWord(lastWord);
+          }
```

Comment:
We should change the logic here a bit. Instead of checking `onKeyUp`, we can hook to the `onInput` event from the input field (add `onInput={this.onInput.bind(this)}` to the input field in `render()`) and when the text has updated, let's save the last word to the state, so that we can pass it as a property to the `EmojiPicker` (more on that above). The input handling would look like below. Notice how we can incorporate the option `useEmojis` here to determine whether we display the component at all.
```javascript
  onInput() {
    const lastWord = this.refs.message.value.split(" ").pop();
    this.setState({ lastWord: lastWord });
    if(lastWord.startsWith(':'))
      this.setState({ emojiPickerActive : this.props.useEmojis });
  }
```